pyccel/codegen/printing/fcode.py

```diff
@@ -699,12 +699,15 @@ def _print_NumpyArray(self, expr):
         rhs_code = self._print(expr.arg)
         # If Numpy array is stored with column-major ordering, transpose values
         # use reshape with order for rank > 2
-        if expr.order == 'F' and expr.rank > 1:
-            shape    = ', '.join(self._print(i) for i in expr.shape)
-            order    = [LiteralInteger(i) for i in range(1, expr.rank+1)]
-            order    = order[1:]+ order[:1]
-            order    = ', '.join(self._print(i) for i in order)
-            rhs_code = 'reshape({},[{}], order=[{}])'.format(rhs_code, shape, order)
+        if expr.order == 'F':
+            if expr.rank == 2:
+                rhs_code = 'transpose({})'.format(rhs_code)
+            elif expr.rank > 2:
+                shape    = ', '.join(self._print(i) for i in expr.shape)
+                order    = [LiteralInteger(i) for i in range(1, expr.rank+1)]
+                order    = order[1:]+ order[:1]
```

Comment:
I would have expected this line to read
```python
order = order[::-1]
```
instead. Have you tested this for `rank > 2`?

Reply:
yes, it's tested for rank > 2, it will give the wrong result if you inverted the order, because the arguments are already treated

Reply:
For example, have you tried something like this?
```python
import numpy as np

x = np.array([[[0, 1, 2, 3], [4, 5, 6, 7], [8, 9, 10, 11]], [[12, 13, 14, 15], [16, 17, 18, 19], [20, 21, 22, 23]]], order='F')
```
Here `x.shape = (2, 3, 4)`.

Reply:
it was tested on something like `np.array([x,y])` where x and y are 2d or 3d arrays

Reply:
done